### DISCUSSION TITLE
How to wrap column data in birt report

### DISCUSSION THREAD

**Discussion by sharath492** [eclipse-birt/birt#1096] [Q&A]
Hi all,

I am using Birt report version 3.2.23.

I have created a report and added a table. The columns attached to the table are pulled from a xml datasource.

The dataset value is dateSetRow["SupplierPartNumber"]="part number 1 45646 45321 4564651 46545 787815 155666 1555555 55555" and for dateSetRow["Description"]= "WWWWWWWWWW WWWWWWWWW WWWWWW WWWWWWWWW WWWWWWWW WWWWW WWWWWWW WWW WWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWWW" like this.

The problem I have is that if string data is added to table cells, I cannot get the text to wrap onto a new line.

I must be missed something obvious. Can anyone please point me in the right direction?

Thanks& regards,
Sharath

[atkorepdf-uk_XR.zip](https://github.com/eclipse/birt/files/9880107/atkorepdf-uk_XR.zip)

**Comment by speckyspooky** [comment]
Hello sharath492,

the solution is to set the "Whitespace"-Attribute of the text/data elements explicit to the value "normal" - see below.
With thus option you will get the linebreak on the preview (HTML) and PDF.

Another hint, you have used a fixed height-value on your table row.
Therefore multi text lines will be display not in full. I removed the height-value completely an so I saw your full text with line break.

![string-break](https://user-images.githubusercontent.com/41593722/198367186-a40e0a94-8f86-4567-8302-e7500435685c.png)

**Comment by sharath492** [comment]
Hi [speckyspooky](https://github.com/speckyspooky),

I removed fixed height value and it worked. 

Thank you so much :)

**Comment by hvbtup** [comment]
As a side note, automatic wrapping of very long words like supercalifragilisticexplialidocious does not work for the PDF emitter if you test your report directly from within the BIRT Designer (they will still be cut off), but you can set a render option when you call BIRT from your Java program.
